### ISSUE TITLE
Support tensor.cumsum() for 1-dim tensors

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#35499]
```
>>> torch.rand(3).cumsum()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cumsum() received an invalid combination of arguments - got (), but expected one of:
 * (name dim, torch.dtype dtype)
 * (int dim, torch.dtype dtype)
```

Currently one must pass `dim=0` or `dim=-1`, which makes code too verbose if one works only with 1-dim tensors

**Comment by colesbury** [comment]
I agree generally, but we should match NumPy's behavior for multi-dim arrays which is to flatten the array if no dim (axis) is given:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.cumsum.html

e.g.

```
x = torch.randn(10, 10)
x.cumsum()  # equivalent to x.reshape(-1).cumsum()
```

**Comment by vadimkantorov** [comment]
This is another case where motivation for this kind of semantics versus explicit `x.reshape(-1).cumsum()` is hard to understand to me. For me this is is quite unintuitive. I wonder how much in-the-wild code uses this default option for multidim.

I guess then my proposal would be enabling parameterless-cumsum for 1-dim and disabling for other cases...

**Comment by vadimkantorov** [comment]
`torch.argsort` has `dim=-1` by default thus supporting 1-dim tensors without any explicit `dim` arguments. Maybe worth making a table checking support of 1-dim tensors.

This would make code more concise
